spec/services/halls/update_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Halls::Update do
+  let!(:hall) { create :hall, name: 'name', capacity: 40 }
+  let(:params) do
+    {
+      name:,
+      capacity: 41
+    }
+  end
+  let(:name) { 'name updated' }
+  let(:instance) { described_class.new(hall.id, params) }
+
+  subject { instance.call }
```

Comment:
```suggestion
  subject(:hall_update) { described_class.new(hall.id, params).call }
```

`instance` it not used in any other place, you can remove it.
Also you can name you subject, then you can read test scenarios like in my other comment.